pkg/api/types.go

```diff
@@ -150,6 +150,11 @@ type ObjectMeta struct {
 	// from the list. If the deletionTimestamp of the object is non-nil, entries
 	// in this list can only be removed.
 	Finalizers []string `json:"finalizers,omitempty"`
+
+	// The cluster which the object belongs to. The federation apiserver/controllers use this to
+	// distinguish the object at the federation level from it's constituents at the cluster level.
+	// An empty value means the object belongs to the local cluster (the one the client connects to).
+	Cluster string `json:"cluster,omitempty"`
```

Comment:
If this refers to a resource by name, typically we would call this "ClusterName"